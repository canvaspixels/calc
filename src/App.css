#root {
  background: #eee;
  display: flex;
  justify-content: center;
  height: 100vh;
  display: flex;
  align-items: center;
}

.app {
  text-align: center;
}

.app-logo {
  height: 80px;
  max-width: 100%;
}

.app-header {
  background-color: #1795d4;
  padding: 20px;
  color: #FFF;
}

.calculator-container {
  display: flex;
  flex-direction: column;
  width: 520px;
  background: #555;
  padding: 0 10px;
}

.calculator-screen {
  flex: 0.5 1;
  display: flex;
  flex-direction: column;
  padding: 10px;
}

.screen-bottom,
.screen-top {
  text-align: right;
  color: #FFF;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  width: 100%;
}

.screen-top {
  font-size: 40px;
  height: 40px;
  align-self: flex-end;
}

.screen-bottom {
  font-size: 120px;
  height: 120px;
  line-height: 1;
  align-self: flex-end;
}

.keypad {
  flex: 1;
  display: flex;
  flex-direction: column;
}

.keypad-key {
  flex: 1;
  display: flex;
}

.btn {
  flex: 1;
  border-radius: 5px;
  margin: 5px;
  padding: 15px 20px 20px;
  line-height: 1;
  border: none;
  background: #000;
  color: #FFF;
  font-size: 40px;
  outline: none;
  cursor: pointer;
}

.btn--zero {
  flex: 2;
  border-radius: 5px;
  align-items: end;
  display: flex;
  padding-left: 50px;
}

.btn--operator {
  background: #1795d4;
}

.btn--misc {
  background: #eee;
  color: #000;
}

@media all and (max-width: 530px) {
  .btn {
    font-size: 32px;
  }

  .calculator-container {
    width: 420px;
  }

  .btn--zero {
    padding-left: 40px;
  }

  .keypad {
    flex: 0.6 0.5;
  }
}

@media all and (max-width: 320px) {
  .btn {
    font-size: 25px;
  }
}
