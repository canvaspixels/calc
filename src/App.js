import React from 'react'
import { connect } from 'react-redux'
import logo from './logo.svg'
import * as evaluateActions from './store/evaluateActions'
import CalculatorScreen from './components/CalculatorScreen'
import Keypad from './components/Keypad'
import './App.css'

export class App extends React.Component {
  render() {
    return (
      <div className="app">
        <header className="app-header">
          <img src={logo} className="app-logo" alt="logo" />
        </header>
        <div className="calculator-container">
          <CalculatorScreen {...this.props} />
          <Keypad {...this.props} />
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => {
  return {
    expression: state.calculator.expression,
    total: state.calculator.total,
  }
}

const mapDispatchToProps = dispatch => {
  return {
    calculate: buttonKey => {
      dispatch(evaluateActions.calculate(buttonKey))
    },
    del: () => {
      dispatch(evaluateActions.deleteLastEntry())
    },
    clear: () => {
      dispatch(evaluateActions.clear())
    },
    evaluate: () => {
      dispatch(evaluateActions.evaluateExpression())
    },
    square: () => {
      dispatch(evaluateActions.square())
    },
    squareRoot: () => {
      dispatch(evaluateActions.squareRoot())
    },
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(App)
