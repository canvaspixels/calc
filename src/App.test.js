import React from 'react'
import { mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import { App } from './App'
import jest from 'jest-mock'

configure({
  adapter: new Adapter(),
})

describe('<App />', () => {
  let wrapper = mount(<App />)

  beforeEach(() => {
    wrapper.setProps({
      evaluate: jest.fn(),
      del: jest.fn(),
      calculate: jest.fn(),
      clear: jest.fn(),
    })
  })

  it('mounts and renders app without errors', () => {
    mount(<App />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('calls the calculate action on keypad number click', () => {
    wrapper.find('button[data-qa="4"]').simulate('click')
    expect(wrapper.prop('calculate').mock.calls.length).toBe(1)
    expect(wrapper.prop('calculate').mock.calls[0][0]).toBe('4')

    expect(wrapper.prop('del').mock.calls.length).toBe(0)
    expect(wrapper.prop('clear').mock.calls.length).toBe(0)
  })

  it('calls del when Del is clicked', () => {
    wrapper.find('button[data-qa="Del"]').simulate('click')
    expect(wrapper.prop('del').mock.calls.length).toBe(1)
  })

  it('calls del when Del is clicked', () => {
    wrapper.find('button[data-qa="Del"]').simulate('click')
    expect(wrapper.prop('del').mock.calls.length).toBe(1)
  })
})
