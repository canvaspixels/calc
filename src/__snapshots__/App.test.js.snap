// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> mounts and renders app without errors 1`] = `
<App
  calculate={[MockFunction]}
  clear={[MockFunction]}
  del={[MockFunction]}
  evaluate={[MockFunction]}
>
  <div
    className="app"
  >
    <header
      className="app-header"
    >
      <img
        alt="logo"
        className="app-logo"
        src="logo.svg"
      />
    </header>
    <div
      className="calculator-container"
    >
      <_default
        calculate={[MockFunction]}
        clear={[MockFunction]}
        del={[MockFunction]}
        evaluate={[MockFunction]}
      >
        <div
          className="calculator-screen"
        >
          <TextFit
            autoResize={true}
            className="screen-top"
            forceSingleModeWidth={true}
            max={40}
            min={1}
            mode="single"
            onReady={[Function]}
            throttle={60}
          >
            <div
              className="screen-top"
              style={
                Object {
                  "fontSize": null,
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "whiteSpace": "nowrap",
                  }
                }
              />
            </div>
          </TextFit>
          <TextFit
            autoResize={true}
            className="screen-bottom"
            forceSingleModeWidth={true}
            max={120}
            min={1}
            mode="single"
            onReady={[Function]}
            throttle={50}
          >
            <div
              className="screen-bottom"
              style={
                Object {
                  "fontSize": null,
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "whiteSpace": "nowrap",
                  }
                }
              />
            </div>
          </TextFit>
        </div>
      </_default>
      <Keypad
        calculate={[MockFunction]}
        clear={[MockFunction]}
        del={[MockFunction]}
        evaluate={[MockFunction]}
      >
        <div
          className="keypad"
        >
          <div
            className="keypad-key"
            key="0"
          >
            <_default
              buttonKey="c"
              key="c"
              onButtonClick={[Function]}
            >
              <button
                className="btn  btn--misc"
                data-qa="c"
                onClick={[Function]}
              >
                c
              </button>
            </_default>
            <_default
              buttonKey="%"
              key="%"
              onButtonClick={[Function]}
            >
              <button
                className="btn  btn--misc"
                data-qa="%"
                onClick={[Function]}
              >
                %
              </button>
            </_default>
            <_default
              buttonKey="Del"
              key="Del"
              onButtonClick={[Function]}
            >
              <button
                className="btn  btn--misc"
                data-qa="Del"
                onClick={[Function]}
              >
                Del
              </button>
            </_default>
            <_default
              buttonKey="/"
              key="/"
              onButtonClick={[Function]}
            >
              <button
                className="btn btn--operator"
                data-qa="/"
                onClick={[Function]}
              >
                /
              </button>
            </_default>
          </div>
          <div
            className="keypad-key"
            key="1"
          >
            <_default
              buttonKey="7"
              key="7"
              onButtonClick={[Function]}
            >
              <button
                className="btn"
                data-qa="7"
                onClick={[Function]}
              >
                7
              </button>
            </_default>
            <_default
              buttonKey="8"
              key="8"
              onButtonClick={[Function]}
            >
              <button
                className="btn"
                data-qa="8"
                onClick={[Function]}
              >
                8
              </button>
            </_default>
            <_default
              buttonKey="9"
              key="9"
              onButtonClick={[Function]}
            >
              <button
                className="btn"
                data-qa="9"
                onClick={[Function]}
              >
                9
              </button>
            </_default>
            <_default
              buttonKey="*"
              key="*"
              onButtonClick={[Function]}
            >
              <button
                className="btn btn--operator"
                data-qa="*"
                onClick={[Function]}
              >
                x
              </button>
            </_default>
          </div>
          <div
            className="keypad-key"
            key="2"
          >
            <_default
              buttonKey="4"
              key="4"
              onButtonClick={[Function]}
            >
              <button
                className="btn"
                data-qa="4"
                onClick={[Function]}
              >
                4
              </button>
            </_default>
            <_default
              buttonKey="5"
              key="5"
              onButtonClick={[Function]}
            >
              <button
                className="btn"
                data-qa="5"
                onClick={[Function]}
              >
                5
              </button>
            </_default>
            <_default
              buttonKey="6"
              key="6"
              onButtonClick={[Function]}
            >
              <button
                className="btn"
                data-qa="6"
                onClick={[Function]}
              >
                6
              </button>
            </_default>
            <_default
              buttonKey="-"
              key="-"
              onButtonClick={[Function]}
            >
              <button
                className="btn btn--operator"
                data-qa="-"
                onClick={[Function]}
              >
                -
              </button>
            </_default>
          </div>
          <div
            className="keypad-key"
            key="3"
          >
            <_default
              buttonKey="1"
              key="1"
              onButtonClick={[Function]}
            >
              <button
                className="btn"
                data-qa="1"
                onClick={[Function]}
              >
                1
              </button>
            </_default>
            <_default
              buttonKey="2"
              key="2"
              onButtonClick={[Function]}
            >
              <button
                className="btn"
                data-qa="2"
                onClick={[Function]}
              >
                2
              </button>
            </_default>
            <_default
              buttonKey="3"
              key="3"
              onButtonClick={[Function]}
            >
              <button
                className="btn"
                data-qa="3"
                onClick={[Function]}
              >
                3
              </button>
            </_default>
            <_default
              buttonKey="+"
              key="+"
              onButtonClick={[Function]}
            >
              <button
                className="btn btn--operator"
                data-qa="+"
                onClick={[Function]}
              >
                +
              </button>
            </_default>
          </div>
          <div
            className="keypad-key"
            key="4"
          >
            <_default
              buttonKey="0"
              key="0"
              onButtonClick={[Function]}
            >
              <button
                className="btn   btn--zero"
                data-qa="0"
                onClick={[Function]}
              >
                0
              </button>
            </_default>
            <_default
              buttonKey="x^2"
              key="x^2"
              onButtonClick={[Function]}
            >
              <button
                className="btn"
                data-qa="x^2"
                onClick={[Function]}
              >
                x^2
              </button>
            </_default>
            <_default
              buttonKey="√"
              key="√"
              onButtonClick={[Function]}
            >
              <button
                className="btn"
                data-qa="√"
                onClick={[Function]}
              >
                √
              </button>
            </_default>
            <_default
              buttonKey="."
              key="."
              onButtonClick={[Function]}
            >
              <button
                className="btn"
                data-qa="."
                onClick={[Function]}
              >
                .
              </button>
            </_default>
            <_default
              buttonKey="="
              key="="
              onButtonClick={[Function]}
            >
              <button
                className="btn btn--operator"
                data-qa="="
                onClick={[Function]}
              >
                =
              </button>
            </_default>
          </div>
        </div>
      </Keypad>
    </div>
  </div>
</App>
`;
